core/javax.cache/src/main/java/org/wso2/carbon/caching/impl/CacheCleanupTask.java

```diff
@@ -58,15 +58,19 @@ public synchronized void run() {
             cc.setTenantId(MultitenantConstants.SUPER_TENANT_ID);
             cc.setTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
             for (CacheImpl cache : caches) {
-                cache.runCacheExpiry();
-                if (log.isDebugEnabled()) {
-                    log.debug("Cache expiry completed for cache " + cache.getName());
+                try {
+                    cache.runCacheExpiry();
+                    if (log.isDebugEnabled()) {
+                        log.debug("Cache expiry completed for cache " + cache.getName());
+                    }
+                } catch (IllegalStateException e) {
+                    if (log.isDebugEnabled()) {
```

Comment:
Since we are consuming the exception, shall we add an error log? We can skip adding the exception in the error log, but in the debug log as this code does.

Reply:
This a scheduled task that will run every 30 seconds. If we add an error log, it will print the error logs unnecessarily.